pkg/controller/runtimeextractor/controller.go

```
@@ -0,0 +1,195 @@
+		config:          configNotifier,
+		updateCh:        updateCh,
+		resourceManager: rm,
+	}
+}
+
+// Run starts the runtime extractor controller and handles configuration changes and version updates.
+// It performs initial deployment based on configuration, then watches for:
+//   - Configuration changes (create/delete resources based on DisableRuntimeExtractor flag)
+//   - Cluster version updates (update DaemonSet images to match new cluster version)
+//
+// The controller runs until the context is cancelled.
```

golangci-lint (2.11.4)

[error] `cancelled` is a misspelling of `canceled`

(misspell)